### ISSUE TITLE
Enhancement: Support PSR-15 middleware adaption

### ISSUE THREAD

**Issue by sminnee** [robbieaverill/psr7-adapters#1]
Now that SS4 has a robust middleware system, create 1 or more middlwareadapters that let you plug in PSR-7 based middleware.

Since there isn't yet a single convention for these middlewares, we might want to have a few, e.g. `SlimMiddlewareAdapter`

**Comment by robbieaverill** [comment]
@sminnee Are you referencing [this proposal](https://github.com/php-fig/fig-standards/blob/master/proposed/http-middleware/middleware.md) for the spec?

**Comment by sminnee** [comment]
I hadn't when I wrote this but that's probably the best bet. The composer package for PSR-15 already exists so you could just tie to that.

**Comment by sminnee** [comment]
It could also be that you do a few different micropackages, so `psr15-middleware` depended on `psr7-adapters`

**Comment by robbieaverill** [comment]
True. Or maybe this could become `psr-adapters` and do all the things

**Comment by robbieaverill** [comment]
Updated (accepted) PSR spec: https://github.com/php-fig/fig-standards/blob/master/accepted/PSR-15-request-handlers.md
